apps/web/vibes/soul/form/file-upload/index.tsx

```diff
@@ -0,0 +1,1244 @@
+
+  if (!shouldRender) return null;
+
+  return (
+    <div
+      className="fade-in-0 slide-in-from-top-2 animate-in mt-3 grid gap-3"
+      id={context.listId}
+      role="list"
+    >
+      {files.map((file, index) => (
+        <FileUploadItem
+          key={`${file.name}-${file.size}-${file.lastModified}-${index}`}
+          value={file}
+        />
+      ))}
+    </div>
+  );
+};
+
+interface FileUploadItemContextValue {
+  id: string;
+  fileState: FileState | undefined;
+  nameId: string;
+  sizeId: string;
+  statusId: string;
+  messageId: string;
+}
+
+const FileUploadItemContext = createContext<FileUploadItemContextValue | null>(null);
+
+function useFileUploadItemContext() {
+  const context = useContext(FileUploadItemContext);
+  if (!context) {
+    throw new Error('File upload item context not found');
+  }
+  return context;
+}
+
+interface FileUploadItemProps extends ComponentPropsWithoutRef<'div'> {
+  value: File;
+}
+
+const FileUploadItem = ({ value }: FileUploadItemProps) => {
+  const id = useId();
+  const statusId = `${id}-status`;
+  const nameId = `${id}-name`;
+  const sizeId = `${id}-size`;
+  const messageId = `${id}-message`;
+
+  const fileState = useStore((state) => state.files.get(value));
+  const fileCount = useStore((state) => state.files.size);
+  const fileIndex = useStore((state) => {
+    const files = Array.from(state.files.keys());
+    return files.indexOf(value) + 1;
+  });
+
+  const itemContext = useMemo(
+    () => ({
+      id,
+      fileState,
+      nameId,
+      sizeId,
+      statusId,
+      messageId,
+    }),
+    [id, fileState, statusId, nameId, sizeId, messageId],
+  );
+
+  if (!fileState) return null;
+
+  let statusText: string;
+
+  if (Boolean(fileState.error)) {
+    statusText = `Error: ${fileState.error}`;
+  } else if (fileState.status === 'uploading') {
+    statusText = `Uploading: ${fileState.progress}% complete`;
+  } else if (fileState.status === 'success') {
+    statusText = 'Upload complete';
+  } else {
+    statusText = 'Ready to upload';
+  }
+
+  return (
+    <FileUploadItemContext.Provider value={itemContext}>
+      <div
+        aria-describedby={`${nameId} ${sizeId} ${statusId} ${Boolean(fileState.error) ? messageId : ''}`}
+        aria-labelledby={nameId}
+        aria-posinset={fileIndex}
+        aria-setsize={fileCount}
+        className={clsx(
+          'relative flex items-center justify-between gap-2 overflow-hidden rounded-lg border-[1.5px] border-(--file-upload-item-border,var(--contrast-200)) p-4',
+          'data-[invalid]:border-(--file-upload-item-border-error,var(--error))',
+        )}
+        data-invalid={itemContext.fileState?.status === 'error' ? '' : undefined}
+        id={id}
+        role="listitem"
+      >
+        <FileUploadItemMetadata />
+        <FileUploadItemDelete />
+        <FileUploadItemProgress />
+        <span className="sr-only" id={statusId}>
+          {statusText}
+        </span>
```

Comment:
This text seems to be generated within the component and I don't see a mechanism for the user of the component to control these, I see that the docs mention internationalization support out of the box for vibes - am I missing something?